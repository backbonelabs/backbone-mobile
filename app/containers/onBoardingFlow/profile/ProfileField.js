import React from 'react';
import {
  View,
  Text,
  TouchableOpacity,
} from 'react-native';
import { map, uniqueId } from 'lodash';
import styles from '../../../styles/onBoarding/profile';

const { PropTypes } = React;

const ProfileField = props => {
  const { height, weight, birthdate: b } = props;
  const dateString = `${b.getMonth() + 1}-${b.getDate()}-${b.getFullYear()}`;
  const formattedProfile = { birthdate: dateString, height, weight };

  return (
    <View style={styles.profileInfoContainer}>
      { map(formattedProfile, (value, key) => (
        <TouchableOpacity
          key={`${value}-${uniqueId()}`}
          style={styles.profileInfoField}
          onPress={() => props.setPickerType(key)}
        >
          <Text style={{ marginLeft: 15 }}>
            {`${key.charAt(0).toUpperCase()}${key.slice(1)}`}
          </Text>
          <Text style={{ marginRight: 15 }}>{value}</Text>
        </TouchableOpacity>
      )) }
    </View>
  );
};

ProfileField.propTypes = {
  setPickerType: PropTypes.func,
  pickerType: PropTypes.string,
  birthdate: PropTypes.object,
  weight: PropTypes.string,
  weightMetric: PropTypes.string,
  height: PropTypes.string,
  heightMetric: PropTypes.string,
};

export default ProfileField;
