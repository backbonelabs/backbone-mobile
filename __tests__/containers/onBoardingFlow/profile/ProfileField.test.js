import React from 'react';
import { shallow } from 'enzyme';
import birthdayIconOff from '../../../../app/images/onboarding/birthday-icon-off.png';
import birthdayIcon from '../../../../app/images/onboarding/birthday-icon-on.png';
import ProfileField from '../../../../app/containers/onBoardingFlow/profile/ProfileField';

describe('ProfileField Component', () => {
  const valEmptyLabel = {
    type: 'birthdate',
    icon: birthdayIcon,
    iconOff: birthdayIconOff,
  };
  const val = {
    ...valEmptyLabel,
    label: '2017-08-01',
  };

  const wrapper = shallow(
    <ProfileField
    	value={val}
    	key={val.type}
    	setPickerType={() => {
        // Empty test function
      }}
      icon={val.icon}
    />
  );
  const wrapperEmptyLabel = shallow(
    <ProfileField
    	value={valEmptyLabel}
    	key={valEmptyLabel.type}
    	setPickerType={() => {
        // Empty test function
      }}
      icon={valEmptyLabel.icon}
    />
  );

  let render = wrapper.dive();

  test('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });

  test('renders a BodyText if label exists', () => {
  	expect(wrapper.find('BodyText')).toHaveLength(1);
  });

  test('renders a SecondaryText if no label found', () => {
    expect(wrapperEmptyLabel.find('SecondaryText')).toHaveLength(1);
  });
});
