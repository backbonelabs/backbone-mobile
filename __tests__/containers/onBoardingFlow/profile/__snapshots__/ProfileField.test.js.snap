// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileField Component renders as expected 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <Image
    source={1}
  />
  <BodyText>
    2017-08-01
  </BodyText>
</View>
`;
